service/http/http.go

```diff
@@ -47,12 +48,26 @@ type Options struct {
 	EnablePProf      bool   // Whether pprof endpoints should be exposed.
 }
 
+// Arguments holds runtime settings for the HTTP service.
+type Arguments struct {
+	TLS *TLSArguments `river:"tls,block,optional"`
+}
+
 type Service struct {
 	log      log.Logger
 	tracer   trace.TracerProvider
 	gatherer prometheus.Gatherer
 	opts     Options
 
+	// lazyLis and lazyNetLis are used to lazily enable TLS.
+	//
+	// lazyNetLis eventually wraps around the real network listener once the
+	// Service is running.
+	//
+	// lazyLis either wraps around a TLS listener or lazyNetLis, depending on
+	// whether TLS is currently enabled.
+	lazyLis, lazyNetLis *lazyListener
```

Comment:
Maybe we can try some different names to make this easier to follow 🤔 
* lazyLis -> publicLis or outerLis
* lazyNetLis -> tcpLis or innerLis